rapids-cmake/cpm/spdlog.cmake

```diff
@@ -62,16 +89,43 @@ function(rapids_cpm_spdlog)
   include("${rapids-cmake-dir}/cpm/detail/generate_patch_command.cmake")
   rapids_cpm_generate_patch_command(spdlog ${version} patch_command)
 
+  # If the option wasn't passed to the command, default to header only fmt
+  if(NOT _RAPIDS_FMT_OPTION)
+    set(_RAPIDS_FMT_OPTION "EXTERNAL_FMT_HO")
+  endif()
+
+  if("${_RAPIDS_FMT_OPTION}" STREQUAL "BUNDLED")
+    set(spdlog_fmt_option "")
+  elseif("${_RAPIDS_FMT_OPTION}" STREQUAL "EXTERNAL_FMT" OR "${_RAPIDS_FMT_OPTION}" STREQUAL
+                                                            "EXTERNAL_FMT_HO")
+    include("${rapids-cmake-dir}/cpm/fmt.cmake")
+
+    # Using `spdlog_ROOT` needs to cause any internal find calls in `spdlog-config.cmake` to first
+    # search beside it before looking globally.
+    list(APPEND fmt_ROOT ${spdlog_ROOT})
+
+    rapids_cpm_fmt(BUILD_EXPORT_SET spdlog INSTALL_EXPORT_SET spdlog)
```

Comment:
I think we want to propagate the user provided build and install export sets. If the user hasn't enabled an install export set for spdlog we shouldn't have one for fmt. ( Would be wrong to install fmt and not spdlog ).

Reply:
My intention with this was to attach fmt as a build and install export to the spdlog target where then if the user added spdlog to a build or install export, fmt would be added transitively. I'm not sure if that's possible, but if it was I'm guessing I should be doing it against the `spdlog::spdlog` and `spdlog::spdlog_header_only` targets instead.

Reply:
That would be possible by passing the same BUILD and EXPORT set down to `fmt` that was provided to `spdlog`.

That will ensure that fmt is installed when spdlog is installed ( and vice versa ) plus will ensure `find_package(fmt)` and `find_package(spdlog)` are performed by the project that is associated with the export set.

Reply:
I believe I addressed this in the way you suggested, but I'm not 100% sure if I did what you were thinking. PTAL

Reply:
You will need to do the following:

```cmake
  set(user_export_set_args)
  if(_RAPIDS_INSTALL_EXPORT_SET)
    list(APPEND user_export_set_args INSTALL_EXPORT_SET ${_RAPIDS_INSTALL_EXPORT_SET})
  endif()
  if(_RAPIDS_BUILD_EXPORT_SET)
    list(APPEND user_export_set_args BUILD_EXPORT_SET ${_RAPIDS_BUILD_EXPORT_SET})
  endif()
  list(APPEND _RAPIDS_UNPARSED_ARGUMENTS ${user_export_set_args})

...

rapids_cpm_fmt(${user_export_set_args})
  ```
  
  That way we properly handle cases where only a BUILD_EXPORT_SET or INSTALL_EXPORT_SET is provided.

Reply:
I had already been updating `_RAPIDS_UNPARSED_ARGUMENTS` like this, so I just reused that variable for passing to `rapids_cpm_fmt`. Should be addressed in the latest commit.

I wasn't sure if the `_RAPIDS_UNPARSED_ARGUMENTS` is used elsewhere, but it didn't look like it.

Reply:
That is an interesting solution. Using `_RAPIDS_UNPARSED_ARGUMENTS` might actually be better as now any CPM options passed to `rapids_cpm_spdlog` are also propagated to `rapids_cpm_fmt` which could be useful.